addons/addon-base-raas/packages/base-raas-services/lib/data-egress/data-egress-service.js

```diff
@@ -0,0 +1,236 @@
+/*
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License").
+ *  You may not use this file except in compliance with the License.
+ *  A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ *  or in the "license" file accompanying this file. This file is distributed
+ *  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+const _ = require('lodash');
+const uuid = require('uuid/v1');
+const { runAndCatch } = require('@aws-ee/base-services/lib/helpers/utils');
+const Service = require('@aws-ee/base-services-container/lib/service');
+const createSchema = require('../schema/create-egress-store.json');
+const {
+  createAllowStatement,
+  getStatementParamsFn,
+  listStatementParamsFn,
+  putStatementParamsFn,
+  updateS3BucketPolicy,
+  addAccountToStatement,
+} = require('../helpers/utils');
+
+const settingKeys = {
+  tableName: 'dbEgressStore',
+  enableEgressStore: 'enableEgressStore',
+  egressStoreBucketName: 'egressStoreBucketName',
+  egressStoreKmsKeyAliasArn: 'egressStoreKmsKeyAliasArn',
+};
+class DataEgressService extends Service {
```

Comment:
Will you be adding unit tests?

Reply:
Unit test are added in commit: 27525e8ad6b21a6a409a35345341b554ecb6310b